plugins/outputs/groundwork/groundwork.go

```diff
@@ -0,0 +1,363 @@
+}
+
+func init() {
+	outputs.Add("groundwork", func() telegraf.Output {
+		return &Groundwork{
+			ResourceTag:         "host",
+			DefaultHost:         "telegraf",
+			DefaultServiceState: string(transit.ServiceOk),
+		}
+	})
+}
+
+func (g *Groundwork) parseMetric(metric telegraf.Metric) (string, transit.DynamicMonitoredService) {
+	resource := g.DefaultHost
+	if value, present := metric.GetTag(g.ResourceTag); present {
+		resource = value
+	}
+
+	service := metric.Name()
+	if value, present := metric.GetTag("service"); present {
+		service = value
+	}
+
+	status := g.DefaultServiceState
+	value, statusPresent := metric.GetTag("status")
+	if validStatus(value) {
+		status = value
+	}
+
+	message := ""
+	if value, present := metric.GetTag("message"); present {
+		message = value
+	}
+
+	unitType := string(transit.UnitCounter)
+	if value, present := metric.GetTag("unitType"); present {
+		unitType = value
+	}
+
+	critical := -1.0
+	value, criticalPresent := metric.GetTag("critical")
+	if criticalPresent {
+		if s, err := strconv.ParseFloat(value, 64); err == nil {
+			critical = s
+		}
+	}
+
+	warning := -1.0
+	value, warningPresent := metric.GetTag("warning")
+	if warningPresent {
+		if s, err := strconv.ParseFloat(value, 64); err == nil {
+			warning = s
+		}
+	}
+
+	lastCheckTime := transit.NewTimestamp()
+	lastCheckTime.Time = metric.Time()
+	serviceObject := transit.DynamicMonitoredService{
+		BaseTransitData: transit.BaseTransitData{
+			Name:  service,
+			Type:  transit.Service,
+			Owner: resource,
+		},
+		Status:           transit.MonitorStatus(status),
+		LastCheckTime:    lastCheckTime,
+		LastPlugInOutput: message,
+		Metrics:          nil,
+	}
+
+	var err error
+	for _, value := range metric.FieldList() {
+		var thresholds []transit.ThresholdValue
+		if warningPresent {
+			thresholds = append(thresholds, transit.ThresholdValue{
+				SampleType: transit.Warning,
+				Label:      value.Key + "_wn",
+				Value: &transit.TypedValue{
+					ValueType:   transit.DoubleType,
+					DoubleValue: warning,
+				},
+			})
+		}
+		if criticalPresent {
+			thresholds = append(thresholds, transit.ThresholdValue{
+				SampleType: transit.Critical,
+				Label:      value.Key + "_cr",
+				Value: &transit.TypedValue{
+					ValueType:   transit.DoubleType,
+					DoubleValue: critical,
+				},
+			})
+		}
+
+		valueType := transit.DoubleType
+		var floatVal float64
+		var stringVal string
+
+		switch value.Value.(type) {
+		case string:
+			valueType = transit.StringType
+			tmpStr := value.Value.(string)
+			stringVal = tmpStr
+		default:
+			floatVal, err = internal.ToFloat64(value.Value)
+			if err != nil {
+				g.Log.Warnf("could not convert %s to float: %v", value.Key, err)
+			}
+		}
+
+		endTime := transit.NewTimestamp()
+		endTime.Time = metric.Time()
+		serviceObject.Metrics = append(serviceObject.Metrics, transit.TimeSeries{
+			MetricName: value.Key,
+			SampleType: transit.Value,
+			Interval: &transit.TimeInterval{
+				EndTime: endTime,
```

Comment:
Seems to be the same as already calculated `lastCheckTime`?
```suggestion
				EndTime: lastCheckTime,
```

Reply:
Fixed